stats/condition_interceptor.go

```diff
@@ -0,0 +1,185 @@
+package stats
+
+import (
+	"errors"
+	"log"
+
+	"github.com/intervention-engine/fhir/models"
+	"github.com/intervention-engine/fhir/server"
+)
+
+// ConditionStatsCreateInterceptor intercepts any new condition resources added to the database
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsCreateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+}
+
+// NewConditionStatsCreateInterceptor returns an initialized instance of a ConditionStatsCreateInterceptor
+func NewConditionStatsCreateInterceptor(pgDataAccess StatsDataAccess, mongoDataAccess server.DataAccessLayer) *ConditionStatsCreateInterceptor {
+	return &ConditionStatsCreateInterceptor{PgDataAccess: pgDataAccess, MongoDataAccess: mongoDataAccess}
+}
+
+// Before is unused
+func (s *ConditionStatsCreateInterceptor) Before(resource interface{}) {}
+
+// After increments disease statistics for given after a new condition resource is created.
+func (s *ConditionStatsCreateInterceptor) After(resource interface{}) {
+	condition, ok := resource.(*models.Condition)
+
+	if ok {
+
+		var err error
+
+		if condition.Subject == nil {
+			log.Printf("ConditionStatsCreateInterceptor: Condition %s has no subject\n", condition.Id)
+			return
+		}
+
+		result, err := s.MongoDataAccess.Get(condition.Subject.ReferencedID, "Patient")
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: Failed to get patient for condition %s\n", condition.Id)
+			return
+		}
+
+		if !conditionIsAbated(condition) {
+			patient := result.(*models.Patient)
+			err = s.PgDataAccess.AddConditionStat(patient, condition)
+		}
+
+		if err != nil {
+			log.Printf("ConditionStatsCreateInterceptor: %s\n", err.Error())
+		}
+	}
+}
+
+// OnError is unused
+func (s *ConditionStatsCreateInterceptor) OnError(err error, resource interface{}) {}
+
+// ConditionStatsUpdateInterceptor intercepts any updates to condition resources in the database
+// and updates the Synthetic Mass condition statistics based on the condition's patient's address.
+type ConditionStatsUpdateInterceptor struct {
+	PgDataAccess    StatsDataAccess
+	MongoDataAccess server.DataAccessLayer
+	// The state of the condition before the database update, for comparison after the database update
+	conditionsBefore map[string]*models.Condition
+	// Tracks if the interceptor failed to cache the condition model before the update
+	cacheError error
```

Comment:
`cacheError` is also susceptible to problems in a multi-threaded environment (since it is a single value shared by multiple threads).  Do you even need it anymore?  If there is an error, then you just won't find the patient in the map when you look for him/her -- so I don't think you need to track an error separately.